zeek/shaper.zql

```diff
@@ -0,0 +1,109 @@
+type snmp={_path:string,ts:time,uid:bstring,id:id,duration:duration,version:bstring,community:bstring,get_requests:uint64,get_bulk_requests:uint64,get_responses:uint64,set_requests:uint64,display_string:bstring,up_since:time,_write_ts:time};
+type socks={_path:string,ts:time,uid:bstring,id:id,version:uint64,user:bstring,password:bstring,status:bstring,request:{host:ip,name:bstring},request_p:port=(uint16),bound:{host:ip,name:bstring},bound_p:port=(uint16),_write_ts:time};
+type software={_path:string,ts:time,host:ip,host_p:port=(uint16),software_type:zenum,name:bstring,version:{major:uint64,minor:uint64,minor2:uint64,minor3:uint64,addl:bstring},unparsed_version:bstring,_write_ts:time};
+type ssh={_path:string,ts:time,uid:bstring,id:id,version:uint64,auth_success:bool,auth_attempts:uint64,direction:zenum,client:bstring,server:bstring,cipher_alg:bstring,mac_alg:bstring,compression_alg:bstring,kex_alg:bstring,host_key_alg:bstring,host_key:bstring,remote_location:{country_code:bstring,region:bstring,city:bstring,latitude:float64,longitude:float64},_write_ts:time};
+type ssl={_path:string,ts:time,uid:bstring,id:id,version:bstring,cipher:bstring,curve:bstring,server_name:bstring,resumed:bool,last_alert:bstring,next_protocol:bstring,established:bool,cert_chain_fuids:[bstring],client_cert_chain_fuids:[bstring],subject:bstring,issuer:bstring,client_subject:bstring,client_issuer:bstring,validation_status:bstring,_write_ts:time};
+type stats={_path:string,ts:time,peer:bstring,mem:uint64,pkts_proc:uint64,bytes_recv:uint64,pkts_dropped:uint64,pkts_link:uint64,pkt_lag:duration,events_proc:uint64,events_queued:uint64,active_tcp_conns:uint64,active_udp_conns:uint64,active_icmp_conns:uint64,tcp_conns:uint64,udp_conns:uint64,icmp_conns:uint64,timers:uint64,active_timers:uint64,files:uint64,active_files:uint64,dns_requests:uint64,active_dns_requests:uint64,reassem_tcp_size:uint64,reassem_file_size:uint64,reassem_frag_size:uint64,reassem_unknown_size:uint64,_write_ts:time};
+type syslog={_path:string,ts:time,uid:bstring,id:id,proto:zenum,facility:bstring,severity:bstring,message:bstring,_write_ts:time};
+type tunnel={_path:string,ts:time,uid:bstring,id:id,tunnel_type:zenum,action:zenum,_write_ts:time};
+type weird={_path:string,ts:time,uid:bstring,id:id,name:bstring,addl:bstring,notice:bool,peer:bstring,source:bstring,_write_ts:time};
+type x509={_path:string,ts:time,id:bstring,certificate:{version:uint64,serial:bstring,subject:bstring,issuer:bstring,not_valid_before:time,not_valid_after:time,key_alg:bstring,sig_alg:bstring,key_type:bstring,key_length:uint64,exponent:bstring,curve:bstring},san:{dns:[bstring],uri:[bstring],email:[bstring],ip:[ip]},basic_constraints:{ca:bool,path_len:uint64},_write_ts:time};
+
+
+put ts = iso(ts), _write_ts=iso(_write_ts) | put .=unflatten(.)
+   | switch (
+    case _path=files => put . = shape(files)
```

Comment:
Order cases by expected frequency for better performance?

Reply:
`zq -f table 'count() by _path | sort -r' zq-sample-data/zng/*` !

Reply:
Performance with/without a switch doesn't vary significantly (at least in my quick and dirty tests, where I compared with just doing `shape(conn)` on a conn log), so I'll leave this optimization for when it does.